addon/-private/system/model/internal-model.js

```diff
@@ -899,8 +899,12 @@ export default class InternalModel {
         }
       }
     });
-    Object.keys(this._implicitRelationships).forEach((key) => {
-      let rel = this._implicitRelationships[key];
+
+    let implicitRelationships = this._implicitRelationships;
+    this.__implicitRelationships = null;
```

Comment:
i think the basic approach is fine and i suspect it seems a bit iffy mainly because these data structures already have a lot of complicated state.  I suspect landing the simplify relationships work will make this kind of resetting not look so bad